agent.go

```diff
@@ -97,13 +97,13 @@ func NewAgent(config *Config, logger *log.Logger) (*Agent, error) {
 		memSink:           memSink,
 	}
 
-	logger.Printf("[INFO] Connecting to Consul on %s...", clientConf.Address)
+	logger.Info("Connecting to Consul", "address", clientConf.Address)
 	for {
 		leader, err := client.Status().Leader()
 		if err != nil {
-			logger.Printf("[ERR] error getting leader status: %q, retrying in %s...", err.Error(), retryTime.String())
+			logger.Error("error getting leader status", "error", hclog.Fmt("%q, retrying in %s...", err.Error(), retryTime.String()))
 		} else if leader == "" {
-			logger.Printf("[INFO] waiting for cluster to elect a leader before starting, will retry in %s...", retryTime.String())
+			logger.Info("waiting for cluster to elect a leader before starting, will retry in ", "time", hclog.Fmt("%s...", retryTime.String()))
```

Comment:
The '...' here doesn't seem very readable now since time is stated in a k/v pair. What do you think about removing it?